t/ui/27-plugin_obs_rsync_batch_status_details.t

```diff
@@ -0,0 +1,192 @@
+my $url = "http://127.0.0.1:$port/public/build/%%PROJECT/_result?package=000product";
+
+$tempdir->child('openqa.ini')->spurt(<<"EOF");
+[global]
+plugins=ObsRsync
+[obs_rsync]
+home=$home
+project_status_url=$url
+EOF
+
+note("Starting fake api server");
+start_server();
+
+my $driver = call_driver(sub { }, {with_gru => 1});
+
+plan skip_all => $OpenQA::SeleniumTest::drivermissing unless $driver;
+
+$driver->find_element_by_class('navbar-brand')->click();
+$driver->find_element_by_link_text('Login')->click();
+
+$driver->get('/admin/obs_rsync/BatchedProj');
+
+is($driver->find_element('tr#folder_Batch1 .project')->get_text(), 'BatchedProj|Batch1', 'Batch name');
+
+is($driver->find_element('tr#folder_Batch1 .lastsync')->get_text(),        '191216_150610', 'Batch last sync');
+is($driver->find_element('tr#folder_Batch1 .lastsyncversion')->get_text(), '470.1',         'Batch synced version');
+# at start no project fetches version from obs
+is($driver->find_element('tr#folder_Batch1 .obsversion')->get_text(), '', 'Batch obs version empty');
+ok(index($driver->find_element('tr#folder_Batch1 .dirtystatuscol .dirtystatus')->get_text(), 'dirty') != -1,
+    'Project dirty status');
+
+# now request fetching version from obs
+$driver->find_element('tr#folder_Batch1 .obsversionupdate')->click();
+wait_for_ajax();
+my $retries = 50;
+my $obsversion;
+while ($retries > 0) {
+    $obsversion = $driver->find_element('tr#folder_Batch1 .obsversion')->get_text();
+    last if $obsversion;
+    sleep(0.1);
+    $retries = $retries - 1;
+}
+
+is($driver->find_element('tr#folder_Batch1 .obsversion')->get_text(), '470.1', 'Batch obs version');
+
+# now we call forget_run_last() and refresh_last_run() and check once again corresponding columns
+$driver->find_element('tr#folder_Batch1 .lastsyncforget')->click();
+$driver->accept_alert;
+
+$retries = 50;
+my $lastsync;
+while ($retries > 0) {
+    $lastsync = $driver->find_element('tr#folder_Batch1 .lastsync')->get_text();
+    last if !$lastsync;
+    sleep(0.1);
+    $retries = $retries - 1;
+}
+
+ok(index($driver->find_element('tr#folder_Batch1 .lastsync')->get_text(), '190703_143010') == -1,
+    'Batch last sync forgotten');
+is($lastsync, '', 'Batch last sync forgotten');
+
+$driver->get('/admin/obs_rsync/BatchedProj');
+is($driver->find_element('tr#folder_Batch1 .lastsync')->get_text(),
+    'no data', 'BatchedProj last sync is still forgotten after refresh');
+
+$driver->find_element('tr#folder_Batch1 .dirtystatusupdate')->click();
+# now wait until gru picks the task up
+$retries = 5;
+my $dirty_status;
+while ($retries > 0) {
+    $dirty_status = $driver->find_element('tr#folder_Batch1 .dirtystatuscol .dirtystatus')->get_text();
+    last if index($dirty_status, 'dirty') == -1;
+    sleep(1);
+    $retries = $retries - 1;
+}
+
+ok(index($dirty_status, 'dirty') == -1,     'BatchedProj dirty status is not dirty anymore');
+ok(index($dirty_status, 'published') != -1, 'BatchedProj dirty is published');
```

Comment:
Using `is`/`isnt` would produce nicer error messages if it fails.

Reply:
Yeah, but it will require additional handling around `$dirty_status`, (because it has timestamp) which itself is error-prone. Will it be ok if I just include `$dirty_status` into the message to simplify eventual troubleshooting?

Reply:
or use: `ok(...) or diag explain $dirty_status;` if it might be long